WIP: Add support for inputtype and make puts public.

marcmerlin opened the pull request:
This does not work as is, but am I on the right track?

It creates this
<div class="box1">Inputtext<form action="/_control?page=p2&item=Inputtext&value=default"><input name="text" value="default" id="p2_i8"/></form></div>

yeah, ok, I need to fix the HTML a bit before it'll work, but would you agree that in m case, I can't use a JS function that auto submits since it's tough to know when an input type is finished before enter and then it sends a http get full page reload (which I'm ok with).

So, should I continue on this track, or do you notice something pretty wrong?
Thanks.

marcmerlin committed:
WIP: Add support for inputtype and make puts public.

marcmerlin committed:
WIP: Add support for inputtype and make puts public.

distrakt commented:
this is exactly how I imagined it yes. And I'm also unsure how to best
indicate completion. Traditionally, "submit" button, and my
handrolled forms do it that way... // d

On Sat, Aug 1, 2020 at 11:07 PM Marc MERLIN <notifications@github.com>
wrote:

> This does not work as is, but am I on the right track?
>
> It creates this
> Inputtext
>
> yeah, ok, I need to fix the HTML a bit before it'll work, but would you
> agree that in m case, I can't use a JS function that auto submits since
> it's tough to know when an input type is finished before enter and then it
> sends a http get full page reload (which I'm ok with).
>
> So, should I continue on this track, or do you notice something pretty
> wrong?
> Thanks.
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/distrakt/OmEspHelpers/pull/6
> Commit Summary
>
>    - WIP: Add support for inputtype and make puts public.
>
> File Changes
>
>    - *M* examples/WebServerSketch1/WebServerSketch1.ino
>    <https://github.com/distrakt/OmEspHelpers/pull/6/files#diff-6fced1238cc197b157dd1c369077206a>
>    (6)
>    - *M* src/OmWebPages.cpp
>    <https://github.com/distrakt/OmEspHelpers/pull/6/files#diff-941c8053d01ddc332ccb7fe97f16436a>
>    (47)
>    - *M* src/OmWebPages.h
>    <https://github.com/distrakt/OmEspHelpers/pull/6/files#diff-e74ecf44183cc82a00aecd4f49bfcd89>
>    (2)
>    - *M* src/OmXmlWriter.h
>    <https://github.com/distrakt/OmEspHelpers/pull/6/files#diff-54dd568d21b355954e09fe35c950098b>
>    (2)
>
> Patch Links:
>
>    - https://github.com/distrakt/OmEspHelpers/pull/6.patch
>    - https://github.com/distrakt/OmEspHelpers/pull/6.diff
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/distrakt/OmEspHelpers/pull/6>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAIYCNLFJKFHHEZN22T47Y3R6T7CDANCNFSM4PSMF55A>
> .
>

marcmerlin commented:
@distrakt  ok, so I got a bit closer, now I have the right URL thanks to hidden form attributes, but I just realized that what I was aiming for is not what I need
So, once I get this http://192.168.205.220/_control?value=defaultmarc&page=p2&item=Inputtext
correctly generated by the submit button, but that doesn't take me to the right page.
I get back "ok" from OmWebPages::handleRequest when instead I need a full page redraw.

Is there an entrypoint to do a full page redraw?
If I'm close and it's simpler for you to finish it than for me through much more trial and error, I won't take offense if you take the patch and fix it/finish it up :)

distrakt commented:
Hi Marc --
The arduino (server) side cannot cause a page refresh... on the webpage
(client) side, look at how function button(button, page, buttonName, v, url) is
implemented, to see how a button-with-link refreshes either the current
location for an empty url, or a page link.

I'll take what you have so far, the new pageitem type. I think there needs
to be a new actionProc type that accepts a string instead of an int, and
the /_control handler can know which to call based on the item type. (Or
could be /_controlString, slightly different url, but same url seems nicer
sort of).

Does that seem to follow?

Thanks for looking at this! But I can't go further for a couple of days, so
feel free to explore further or not as you see fit. :)

// david

P.S. I'm replying by email and hopefully this ends up in the github
sequence, i have not learned all yet.

On Sun, Aug 2, 2020 at 3:57 PM Marc MERLIN <notifications@github.com> wrote:

> @distrakt <https://github.com/distrakt> ok, so I got a bit closer, now I
> have the right URL thanks to hidden form attributes, but I just realized
> that what I was aiming for is not what I need
> So, once I get this
> http://192.168.205.220/_control?value=defaultmarc&page=p2&item=Inputtext
> correctly generated by the submit button, but that doesn't take me to the
> right page.
> I get back "ok" from OmWebPages::handleRequest when instead I need a full
> page redraw.
>
> Is there an entrypoint to do a full page redraw?
> If I'm close and it's simpler for you to finish it than for me through
> much more trial and error, I won't take offense if you take the patch and
> fix it/finish it up :)
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/distrakt/OmEspHelpers/pull/6#issuecomment-667735272>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAIYCNLYSAHYVQHD6KABDH3R6XVN3ANCNFSM4PSMF55A>
> .
>

marcmerlin commented:
Email worked fine. If you want to see the github side, go to
https://github.com/distrakt/OmEspHelpers/pull/6

yes, it does make sense, I kind of got that when I realized where I was, but then I got blocked because I don't know how to make an HTML widget that sends a string without doing a page reload given that my HTML knowledge stops before the AJAX days started.

marcmerlin commented:
Sorry that I suck at JS/AJAX, I started reading up on it a bit, but it looks like a non trivial time investment. I'll go back to what I know better and leave the hard work to you :)

marcmerlin committed:
Memory free on ESP32 was wrong with PSRAM.

It was also very incomplete. ESP32 has 32bit and 8bit memory that
aren't usable for the same things (some things require 8bit memory
others will work with 32bit memory too).

This is what it looks like now:
requests:    1
maxHtml:     0
uptime:      1m4s
32-bit Mem:  95544
32-bit Block:51288
8b/DMA Mem:  52996
8b/DMA Block:51288
PSRAM used:  79148
PSRAM free:  4113720
chipId:      ESP32 @240
clientIp:    192.168.205.4:52842
serverIp:    192.168.205.178:80
wifiNetwork: magicnet-guest
built:       Dec 17 2020 18:27:17

Before the patch, it showed me that I had 4MB of memory free,
which was pretty useless :)

marcmerlin committed:
Make display elements wider to allow for my wide dropdown menus.

marcmerlin commented on src/OmXmlWriter.h:
@distrakt  could I ask you to simply merge this one line change (not the rest of the patch) so that puts can be public?
I do need it public to make my own constructs that you don't support (which is fine) and basically cannot switch back to your unmodified code until this one line is merged.

marcmerlin committed:
Wifi compat with ESP32 2.0 hardware defs.

marcmerlin committed:
Merge pull request #1 from marcmerlin/widerDisplay

Wider display + newer ESP32 hardware defs + PSRAM support

marcmerlin committed:
Terrible patch to support old and new SYSTEM_EVENT_AP_STACONNECTED
